--- !ditz.rubyforge.org,2008-03-06/issue 
title: "KeyError: None exception in line 124 of _format_dict"
desc: |-
  mitmanek:json_diff (master) $ json_diff -H -x time_elapsed -x time /tmp/piglit/main test/new-testing-data.json >diff.html
  Traceback (most recent call last):
    File "/home/matej/.bin/json_diff", line 281, in <module>
      print HTMLFormatter(diff_res)
    File "/home/matej/.bin/json_diff", line 133, in __str__
      return self._generate_page(self.diff).encode("utf-8")
    File "/home/matej/.bin/json_diff", line 76, in _generate_page
      self._format_dict(in_dict))
    File "/home/matej/.bin/json_diff", line 115, in _format_dict
      out_str += self._format_dict(diff_dict[typechange], typechange, level)
    File "/home/matej/.bin/json_diff", line 127, in _format_dict
      out_str += self._format_dict(diff_dict[variable], None, level+1)
    File "/home/matej/.bin/json_diff", line 115, in _format_dict
      out_str += self._format_dict(diff_dict[typechange], typechange, level)
    File "/home/matej/.bin/json_diff", line 127, in _format_dict
      out_str += self._format_dict(diff_dict[variable], None, level+1)
    File "/home/matej/.bin/json_diff", line 124, in _format_dict
      (level_str, STYLE_MAP[typch], variable, unicode(diff_dict[variable])))
  KeyError: None
type: :bugfix
component: json_diff
release: 
reporter: Matej Cepl <mcepl@redhat.com>
status: :closed
disposition: :fixed
creation_time: 2011-11-07 09:58:56.794804 Z
references: []

id: 5f4f6fef53a0907e3565ff2e6888fa916fe463e4
log_events: 
- - 2011-11-07 09:58:58.230693 Z
  - Matej Cepl <mcepl@redhat.com>
  - created
  - ""
- - 2011-11-08 18:11:14.760985 Z
  - Matej Cepl <mcepl@redhat.com>
  - closed with disposition fixed
  - ""
